lib/translator/status.cc

```diff
@@ -0,0 +1,252 @@
+      result.status = scsi_defs::Status::kCheckCondition;
+      result.sense_key = scsi_defs::SenseKey::kMediumError;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kAbortedPowerLoss:
+      result.status = scsi_defs::Status::kTaskAborted;
+      result.sense_key = scsi_defs::SenseKey::kAbortedCommand;
+      result.asc = scsi_defs::AdditionalSenseCode::kWarningPowerLossExpected;
+      result.ascq =
+          scsi_defs::AdditionalSenseCodeQualifier::kWarningPowerLossExpected;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kInternalDeviceError:
+      result.status = scsi_defs::Status::kCheckCondition;
+      result.sense_key = scsi_defs::SenseKey::kHardwareError;
+      result.asc = scsi_defs::AdditionalSenseCode::kInternalTargetFailure;
+      result.ascq =
+          scsi_defs::AdditionalSenseCodeQualifier::kInternalTargetFailure;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kAbortedByRequest:
+      result.status = scsi_defs::Status::kTaskAborted;
+      result.sense_key = scsi_defs::SenseKey::kAbortedCommand;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kAbortedSqDeletion:
+      result.status = scsi_defs::Status::kTaskAborted;
+      result.sense_key = scsi_defs::SenseKey::kAbortedCommand;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kAbortedFailedFused:
+      result.status = scsi_defs::Status::kTaskAborted;
+      result.sense_key = scsi_defs::SenseKey::kAbortedCommand;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kAbortedMissingFused:
+      result.status = scsi_defs::Status::kTaskAborted;
+      result.sense_key = scsi_defs::SenseKey::kAbortedCommand;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kInvalidNamespaceOrFormat:
+      result.status = scsi_defs::Status::kCheckCondition;
+      result.sense_key = scsi_defs::SenseKey::kIllegalRequest;
+      result.asc =
+          scsi_defs::AdditionalSenseCode::kAccessDeniedInvalidLuIdentifier;
+      result.ascq = scsi_defs::AdditionalSenseCodeQualifier::
+          kAccessDeniedInvalidLuIdentifier;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kLbaOutOfRange:
+      result.status = scsi_defs::Status::kCheckCondition;
+      result.sense_key = scsi_defs::SenseKey::kIllegalRequest;
+      result.asc = scsi_defs::AdditionalSenseCode::kLbaOutOfRange;
+      result.ascq = scsi_defs::AdditionalSenseCodeQualifier::kLbaOutOfRange;
+      break;
+    case nvme_defs::GenericCommandStatusCode::kNamespaceNotReady:
+      result.status = scsi_defs::Status::kCheckCondition;
+      result.sense_key = scsi_defs::SenseKey::kNotReady;
+      result.asc = scsi_defs::AdditionalSenseCode::
+          kLogicalUnitNotReadyCauseNotReportable;
+      result.ascq = scsi_defs::AdditionalSenseCodeQualifier::
+          kLogicalUnitNotReadyCauseNotReportable;
+      break;
+    default:
+      DebugLog("No SCSI translation for NVMe status with code %#x\n",
```

Comment:
Let's omit trailing newlines from `DebugLog` invocations; downstream code can add them when appropriate, which isn't always.
```suggestion
      DebugLog("No SCSI translation for NVMe status with code %#x",
```